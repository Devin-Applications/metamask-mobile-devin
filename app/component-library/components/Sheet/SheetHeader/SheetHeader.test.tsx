// Third party dependencies.
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';

// Internal dependencies.
import SheetHeader from './SheetHeader';
import {
  SHEET_HEADER_ACTION_BUTTON_ID,
  SHEET_HEADER_BACK_BUTTON_ID,
} from './SheetHeader.constants';

// Mock clipboard
jest.mock('@react-native-clipboard/clipboard', () => ({
  setString: jest.fn(),
  getString: jest.fn(),
}));

describe('SheetHeader', () => {
  it('should render correctly', () => {
    const { toJSON } = render(<SheetHeader title={'Title'} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render back button', () => {
    render(<SheetHeader onBack={jest.fn} title={'Title'} />);
    const backButton = screen.getByTestId(SHEET_HEADER_BACK_BUTTON_ID);
    expect(backButton).toBeTruthy();
  });

  it('should render action button', () => {
    render(
      <SheetHeader
        title={'Title'}
        actionButtonOptions={{ label: 'Action', onPress: jest.fn }}
      />,
    );
    const actionButton = screen.getByTestId(SHEET_HEADER_ACTION_BUTTON_ID);
    expect(actionButton).toBeTruthy();
  });

  it('should render the correct title', () => {
    render(<SheetHeader title={'Test Title'} />);
    expect(screen.getByText('Test Title')).toBeTruthy();
  });

  it('should call onBack when back button is pressed', () => {
    const onBackMock = jest.fn();
    render(<SheetHeader onBack={onBackMock} title={'Title'} />);
    fireEvent.press(screen.getByTestId(SHEET_HEADER_BACK_BUTTON_ID));
    expect(onBackMock).toHaveBeenCalled();
  });

  it('should call onPress when action button is pressed', () => {
    const onPressMock = jest.fn();
    render(
      <SheetHeader
        title={'Title'}
        actionButtonOptions={{ label: 'Action', onPress: onPressMock }}
      />,
    );
    fireEvent.press(screen.getByTestId(SHEET_HEADER_ACTION_BUTTON_ID));
    expect(onPressMock).toHaveBeenCalled();
  });
});
