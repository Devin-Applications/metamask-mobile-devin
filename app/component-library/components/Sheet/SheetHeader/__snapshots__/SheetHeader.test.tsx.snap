// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SheetHeader should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flexDirection": "row",
      "height": 32,
      "justifyContent": "space-between",
      "margin": 16,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
      }
    }
  />
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#141618",
        "fontFamily": "Euclid Circular B",
        "fontSize": 18,
        "fontWeight": "700",
        "letterSpacing": 0,
        "lineHeight": 24,
      }
    }
  >
    Title
  </Text>
  <View
    style={
      {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  />
</View>
`;
